src/appshell/view/menumodel.cpp

```diff
@@ -0,0 +1,513 @@
+//=============================================================================
+//  MuseScore
+//  Music Composition & Notation
+//
+//  Copyright (C) 2020 MuseScore BVBA and others
+//
+//  This program is free software; you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License version 2.
+//
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU General Public License for more details.
+//
+//  You should have received a copy of the GNU General Public License
+//  along with this program; if not, write to the Free Software
+//  Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+//=============================================================================
+#include "menumodel.h"
+
+#include "translation.h"
+
+using namespace mu::appshell;
+using namespace mu::uicomponents;
+using namespace mu::notation;
+using namespace mu::workspace;
+using namespace mu::actions;
+
+MenuModel::MenuModel(QObject* parent)
+    : QObject(parent)
+{
+}
+
+void MenuModel::load()
+{
+    m_items.clear();
+    m_items << fileItem()
+            << editItem()
+            << viewItem()
+            << addItem()
+            << formatItem()
+            << toolsItem()
+            << helpItem();
+
+    globalContext()->currentMasterNotationChanged().onNotify(this, [this]() {
+        load();
+
+        currentMasterNotation()->notation()->notationChanged().onNotify(this, [this]() {
+            load();
+        });
+    });
+
+    emit itemsChanged();
+}
+
+void MenuModel::handleAction(const QString& actionCode)
+{
+    actionsDispatcher()->dispatch(actionCode.toStdString());
+}
+
+QVariantList MenuModel::items()
+{
+    QVariantList menuItems;
+
+    for (const MenuItem& menuItem: m_items) {
+        menuItems << menuItem.toVariantMap();
+    }
+
+    return menuItems;
+}
+
+IMasterNotationPtr MenuModel::currentMasterNotation() const
+{
+    return globalContext()->currentMasterNotation();
+}
+
+INotationPtr MenuModel::currentNotation() const
+{
+    return currentMasterNotation() ? currentMasterNotation()->notation() : nullptr;
+}
+
+MenuItem MenuModel::fileItem()
+{
+    MenuItemList fileItems {
+        makeAction(actionsRegister()->action("file-new")),
```

Comment:
Look like better make `makeAction("file-new")` and call `actionsRegister()->action(...)` inside `makeAction`